enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -980,19 +980,32 @@ class AdjointGenerator
       break;
     }
     case DerivativeMode::ForwardMode: {
-      Value *orig_op0 = I.getOperand(0);
-
       IRBuilder<> Builder2(&I);
       getForwardBuilder(Builder2);
 
+      Value *orig_op0 = I.getOperand(0);
+      Type *diffTy = gutils->getWidth() > 1
+                         ? gutils->getTypeForVectorMode(I.getType())
+                         : I.getType();
+
+      auto rule = [&](Value *dif) {
+        return Builder2.CreateCast(I.getOpcode(), dif, I.getType());
+      };
+
       if (!gutils->isConstantValue(orig_op0)) {
-        Value *dif = diffe(orig_op0, Builder2);
-        setDiffe(&I, Builder2.CreateCast(I.getOpcode(), dif, I.getType()),
-                 Builder2);
+        Value *dop0 = diffe(orig_op0, Builder2);
+
+        Value *diff;
+        if (gutils->getWidth() > 1) {
+          diff = unwrapAndApply(I.getType(), dop0, Builder2, rule);
```

Comment:
It would be nicer as an API here if unwrapAndApply would internally do the width check and decision.

Reply:
sounds reasonable

Reply:
done